scripts/prevent-idle-timeout.sh

```diff
@@ -8,27 +8,30 @@
 # SPDX-License-Identifier: EPL-2.0
 #
 
-# Ensure that main environment variable are configured
-if [ -z "${CHE_API_INTERNAL}" ]; then
-    echo "CHE_API_INTERNAL is not configured in environment."
+CA_CERT="/tmp/che/secret/ca.crt"
+
+
+if [ -z "${CHE_API_INTERNAL}" ] || [ -z "${CHE_WORKSPACE_ID}" ]; then
+    echo "CHE_API_INTERNAL or CHE_WORKSPACE_ID is not configured in environment."
     exit 1
 fi
-if [ -z "${CHE_MACHINE_TOKEN}" ]; then
-    echo "CHE_MACHINE_TOKEN is not configured in environment."
-    exit 1
+
+CURL="curl -X PUT"
+
+if [ -e $CA_CERT ]; then
+	CURL="$CURL --cacert /tmp/che/secret/ca.crt"
 fi
-if [ -z "${CHE_WORKSPACE_ID}" ]; then
-    echo "CHE_WORKSPACE_ID is not configured in environment."
-    exit 1
+
+if [ -n "$CHE_MACHINE_TOKEN" ]; then
+	CURL="$CURL -H \"Authorization: Bearer $CHE_MACHINE_TOKEN\""
 fi
 
-# Todo(vzhukovs): check for client certificates that might be included into request in case if script runs on minikube with self-signed certificates
+CURL="$CURL $CHE_API_INTERNAL/activity/$CHE_WORKSPACE_ID"
 
-# Start making requests to avoid idling, make a request to activity tracker each minute
 while true; do
-    curl -X PUT \
-    -H "Authorization: Bearer ${CHE_MACHINE_TOKEN}" \
-    "${CHE_API_INTERNAL}/activity/${CHE_WORKSPACE_ID}"
+	echo "$(date) - Perform request: '$CURL'."
+	eval "$CURL"
```

Comment:
I think it's less readable than having
CURL_OPTIONS (and adding --cacert) and authorization bearer

and still doing
```
$ curl -X PUT ${CURL_OPTIONS} "${CHE_API_INTERNAL}/activity/${CHE_WORKSPACE_ID}"
```

Reply:
I've tried this approach, but adding `set -x` before curl execution shows:
```
+ curl -X PUT --cacert /tmp/che/secret/ca.crt -H '"Authorization:' Bearer 'token"' http://example.com/activity/wsId
curl: (6) Could not resolve host: Bearer
curl: (6) Could not resolve host: token"
```


<details>
<summary>with this sample of script</summary>

```bash
#!/bin/sh
#
# Copyright (c) 2020 Red Hat, Inc.
# This program and the accompanying materials are made
# available under the terms of the Eclipse Public License 2.0
# which is available at https://www.eclipse.org/legal/epl-2.0/
#
# SPDX-License-Identifier: EPL-2.0
#

CA_CERT="/tmp/che/secret/ca.crt"


if [ -z "${CHE_API_INTERNAL}" ] || [ -z "${CHE_WORKSPACE_ID}" ]; then
    echo "CHE_API_INTERNAL or CHE_WORKSPACE_ID is not configured in environment."
    exit 1
fi

CURL_OPTS=

if [ -e $CA_CERT ]; then
	CURL_OPTS="$CURL_OPTS"' --cacert /tmp/che/secret/ca.crt'
fi

if [ -n "$CHE_MACHINE_TOKEN" ]; then
	CURL_OPTS="$CURL_OPTS"' -H "Authorization: Bearer '"$CHE_MACHINE_TOKEN"'"'
fi

REQUEST_URL="$CHE_API_INTERNAL"'/activity/'"$CHE_WORKSPACE_ID"

while true; do
	echo "$(date) - Perform request: '$CURL_OPTS'."
	set -x
	curl -X PUT $CURL_OPTS $REQUEST_URL
	echo "$(date) - Sleep for 1 minute."

    sleep 60
done
```

</details>

and:
```
+ curl -X PUT ' --cacert /tmp/che/secret/ca.crt -H "Authorization: Bearer token"' http://example.com/activity/wsId
curl: (3) URL using bad/illegal format or missing URL
```

<details>
  <summary>when expanding curl opts</summary>
  
  ```bash
#!/bin/sh
#
# Copyright (c) 2020 Red Hat, Inc.
# This program and the accompanying materials are made
# available under the terms of the Eclipse Public License 2.0
# which is available at https://www.eclipse.org/legal/epl-2.0/
#
# SPDX-License-Identifier: EPL-2.0
#

CA_CERT="/tmp/che/secret/ca.crt"


if [ -z "${CHE_API_INTERNAL}" ] || [ -z "${CHE_WORKSPACE_ID}" ]; then
    echo "CHE_API_INTERNAL or CHE_WORKSPACE_ID is not configured in environment."
    exit 1
fi

CURL_OPTS=

if [ -e $CA_CERT ]; then
	CURL_OPTS="$CURL_OPTS"' --cacert /tmp/che/secret/ca.crt'
fi

if [ -n "$CHE_MACHINE_TOKEN" ]; then
	CURL_OPTS="$CURL_OPTS"' -H "Authorization: Bearer '"$CHE_MACHINE_TOKEN"'"'
fi

REQUEST_URL="$CHE_API_INTERNAL"'/activity/'"$CHE_WORKSPACE_ID"

while true; do
	echo "$(date) - Perform request: '$CURL_OPTS'."
	set -x
	curl -X PUT "$CURL_OPTS" $REQUEST_URL
	echo "$(date) - Sleep for 1 minute."

    sleep 60
done

  ```
  
</details>

Reply:
But with original code I can say, that workspace on CRW 2.5 GA instance was started yesterday at ~10pm and it still active till now.

Reply:
Also PR was checked on che.openshift.io and minikube in the meantime.

Reply:
@vzhukovs I think you're just missing the eval on curl command

```bash
#!/bin/bash
#
# Copyright (c) 2020 Red Hat, Inc.
# This program and the accompanying materials are made
# available under the terms of the Eclipse Public License 2.0
# which is available at https://www.eclipse.org/legal/epl-2.0/
#
# SPDX-License-Identifier: EPL-2.0
#

CA_CERT="/tmp/che/secret/ca.crt"


if [ -z "${CHE_API_INTERNAL}" ] || [ -z "${CHE_WORKSPACE_ID}" ]; then
    echo "CHE_API_INTERNAL or CHE_WORKSPACE_ID is not configured in environment."
    exit 1
fi

CURL_OPTS=""

if [ -e "${CA_CERT}" ]; then
  CURL_OPTS="$CURL_OPTS --cacert /tmp/che/secret/ca.crt"
fi

if [ -n "$CHE_MACHINE_TOKEN" ]; then
  CURL_OPTS="$CURL_OPTS -H \"Authorization: Bearer ${CHE_MACHINE_TOKEN}\""
fi

REQUEST_URL="${CHE_API_INTERNAL}/activity/${CHE_WORKSPACE_ID}"

while true; do
  echo "$(date) - Perform request: '$CURL_OPTS'."
  set -x
  eval curl -X PUT "${CURL_OPTS}" "${REQUEST_URL}"
  echo "$(date) - Sleep for 1 minute."
  sleep 60
done

```

Reply:
Much appreciated for your help! Now it works!